### PR TITLE
chore(mypy): add type checking for python 2

### LINKED ISSUE #3845: mypy invalid syntax error with py2
### Which version of dd-trace-py are you using?

v1.2.0

### Which version of pip are you using?

22.1.2

### Which version of the libraries are you using?

```
ddtrace==1.2.0
mypy[python2]==0.961
```

### How can we reproduce your problem?

1. Clone https://gist.github.com/majorgreys/720201c4bdd33131e4c231207ef716ce
2. Run `docker run --rm -it (docker build .)`

### What is the result that you get?

```
usr/local/lib/python3.9/site-packages/ddtrace/contrib/redis/asyncio_patch.py:12: error: invalid syntax
Found 1 error in 1 file (errors prevented further checking)
```

### What is the result that you expected?

```
Success: no issues found in 1 source file
```

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

Related to #3845 

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: mypy_py2.ini [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+show_error_codes = true
+show_error_context = true
+warn_unused_ignores = true
+warn_unused_configs = true
+no_implicit_optional = true
+ignore_missing_imports = true
+namespace_packages = true
+
+[mypy-ddtrace.contrib.*]
+ignore_errors = true
```
[the remark this answers is not in the document]
Comment (majorgreys) [reply mypy_py2.ini:20]: Yes, we do but that should be handled separately than this.

File: ddtrace/contrib/redis/patch.py [hunk 2]
```diff
@@ -36,14 +37,15 @@ def patch():
...
         _w("redis.client", "Pipeline.immediate_execute_command", traced_execute_command)
-        if redis.VERSION >= (4, 2, 0):
-            from .asyncio_patch import traced_async_execute_command
-            from .asyncio_patch import traced_async_execute_pipeline
-
-            _w("redis.asyncio.client", "Redis.execute_command", traced_async_execute_command)
-            _w("redis.asyncio.client", "Redis.pipeline", traced_pipeline)
-            _w("redis.asyncio.client", "Pipeline.execute", traced_async_execute_pipeline)
-            _w("redis.asyncio.client", "Pipeline.immediate_execute_command", traced_async_execute_command)
+        if PY3:
```
Comment (mabdinur) [inline ddtrace/contrib/redis/patch.py:40]: This change might conflict with this PR: https://github.com/DataDog/dd-trace-py/pull/3868. Should we revert this change?

File: ddtrace/contrib/redis/patch.py [hunk 3]
```diff
@@ -62,6 +64,7 @@ def unpatch():
             unwrap(redis.Redis, "pipeline")
             unwrap(redis.client.Pipeline, "execute")
             unwrap(redis.client.Pipeline, "immediate_execute_command")
+        if PY3:
             if redis.VERSION >= (4, 2, 0):
```
Comment (mabdinur) [inline ddtrace/contrib/redis/patch.py:68]: Do we need a python 3 check here?
